tools/firely-auth/firely-auth-settings.rst

```diff
@@ -293,11 +293,208 @@ You register a :term:`client` in the ``AllowedClients`` array. For each client y
+          "LaunchIds": ["<same id as for user>"],
+          "RefreshTokenLifetime": "90"
+        },
+        {
+          "ClientId": "Inferno-Public",
+          "ClientName": "InfernoPublic",
+          "Enabled": true,
+          "RequireConsent": true,
+          "RedirectUris": [ "https://inferno.healthit.gov/suites/custom/smart/launch", "https://inferno.healthit.gov/suites/custom/smart/redirect"],
+          "AllowedGrantTypes": [ "authorization_code" ],
+          "AllowFirelySpecialScopes": false,
+          "AllowedSmartLegacyActions": [ "read", "write", "*" ],
+          "AllowedSmartActions": [ "c", "r", "u", "d", "s" ],
+          "AllowedSmartSubjects": [ "patient", "user" ],
+          "AlwaysIncludeUserClaimsInIdToken": true,
+          "RequirePkce": false,
+          "AllowOfflineAccess": true,
+          "AllowOnlineAccess": false,
+          "LaunchIds": [],
+          "RequireClientSecret": false,
+          "RefreshTokenLifetime": "90"
+        },
+        {
+          "ClientId": "Inferno-Bulk",
+          "ClientName": "InfernoBulk",
+          "Enabled": true,
+          "RedirectUris": [ "https://inferno.healthit.gov/suites/custom/smart/launch", "https://inferno.healthit.gov/suites/custom/smart/redirect"],
+          "AllowedGrantTypes": [ "authorization_code", "client_credentials" ],
+          "AllowFirelySpecialScopes": false,
+          "AllowedSmartLegacyActions": [ "read" ],
+          "AllowedSmartActions": [ "c", "r", "u", "d", "s" ],
+          "AllowedSmartSubjects": [ "system" ],
+          "RequirePkce": false,
+          "AllowOfflineAccess": true,
+          "AllowOnlineAccess": false,
+          "ClientSecrets": [
+            {
+              "SecretType": "JWK",
+              "Secret": "{'e':'AQAB','kid':'b41528b6f37a9500edb8a905a595bdd7','kty':'RSA','n':'vjbIzTqiY8K8zApeNng5ekNNIxJfXAue9BjoMrZ9Qy9m7yIA-tf6muEupEXWhq70tC7vIGLqJJ4O8m7yiH8H2qklX2mCAMg3xG3nbykY2X7JXtW9P8VIdG0sAMt5aZQnUGCgSS3n0qaooGn2LUlTGIR88Qi-4Nrao9_3Ki3UCiICeCiAE224jGCg0OlQU6qj2gEB3o-DWJFlG_dz1y-Mxo5ivaeM0vWuodjDrp-aiabJcSF_dx26sdC9dZdBKXFDq0t19I9S9AyGpGDJwzGRtWHY6LsskNHLvo8Zb5AsJ9eRZKpnh30SYBZI9WHtzU85M9WQqdScR69Vyp-6Uhfbvw'}"
+            },
+            {
+              "SecretType": "JWK",
+              "Secret": "{'kty':'EC','crv':'P-384','x':'JQKTsV6PT5Szf4QtDA1qrs0EJ1pbimQmM2SKvzOlIAqlph3h1OHmZ2i7MXahIF2C','y':'bRWWQRJBgDa6CTgwofYrHjVGcO-A7WNEnu4oJA5OUJPPPpczgx1g2NsfinK-D2Rw','key_ops':['verify'],'ext':true,'kid':'4b49a739d1eb115b3225f4cf9beb6d1b','alg':'ES384'}"
+            }
+          ],
+          "RequireClientSecret": true,
+          "RefreshTokenLifetime": "90"
+        }
+      ]
+    }
+  }
+
+Firely Server settings
+^^^^^^^^^^^^^^^^^^^^^^
+
+Put these settings in appsettings.instance.json, next to the executable.
+
+.. code-block:: json
+
+  "SmartAuthorizationOptions": {
+    "Enabled": true,
+    "Filters": [
+      {
+        "FilterType": "Patient",
+        "FilterArgument": "_id=#patient#"
```

Comment:
```suggestion
        "FilterArgument": "_id=#patient#" // "identifier=#patient#", this setting should be correspondent with Firely Auth
```

Reply:
See above, I won't elaborate on the options for using Filters,